src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```diff
@@ -850,12 +851,10 @@ private void clone(Run<?, ?> build, Launcher launcher, FilePath repository, Node
         if (cachedSource != null && !cachedSource.isUseSharing()) {
             FilePath hgrc = repository.child(".hg/hgrc");
             if (hgrc.exists()) {
-                String hgrcText = hgrc.readToString();
-                if (!hgrcText.contains(cachedSource.getRepoLocation())) {
-                    listener.error(".hg/hgrc did not contain " + cachedSource.getRepoLocation() + " as expected:\n" + hgrcText);
-                    throw new AbortException(".hg/hgrc did not contain " + cachedSource.getRepoLocation() + " as expected:\n" + hgrcText);
-                }
-                hgrc.write(hgrcText.replace(cachedSource.getRepoLocation(), getSource(env)), null);
+                Ini hgrcIni = new Ini(hgrc.read());
```

Comment:
File handle leak.
```suggestion
                try (InputStream is = hgrc.read()) {
                    Ini hgrcIni = new Ini(is);
```
(plus closing brace—GitHub does not let me suggest that!)

Reply:
In fact I was worried about potential leaks, but relied to much on Findbugs to reveal them to me.